packages/gitmoji-changelog-markdown/src/index.js

```diff
@@ -31,51 +32,59 @@ function toMarkdown({ meta, changes }) {
 }
 
 function markdownFromScratch({ meta, changes }, options) {
-  const output = toMarkdown({ meta, changes })
-  return Promise.resolve(fs.writeFileSync(options.output, output))
+  return promisify(fs.writeFile)(options.output, `# Changelog\n\n${toMarkdown({ meta, changes })}`)
 }
 
 function markdownIncremental({ meta, changes }, options) {
   const { lastVersion } = meta
-
-  const currentFile = options.output
-  const tempFile = `${currentFile}.tmp`
-
-  if (!fs.existsSync(currentFile)) {
-    throw new Error(`${currentFile} doesn't exists, please execute "gitmoji-changelog init" to build it from scratch.`)
-  }
-
-  // write file for next version
-  const writer = fs.createWriteStream(tempFile, { encoding: 'utf-8' })
-  writer.write(toMarkdown({ meta, changes }))
-
-  // read original file until last tags and add it to the end
-  let lastVersionFound = false
-  return new Promise(resolve => {
-    const stream = fs.createReadStream(currentFile)
-      .pipe(es.split())
-      .pipe(es.mapSync((line) => {
-        stream.pause()
-
-        if (line.startsWith(`<a name="${lastVersion}"></a>`)) {
-          lastVersionFound = true
-        }
-        if (lastVersionFound) {
-          writer.write(`${line}\n`)
-        }
-
-        stream.resume()
+  const { output, release } = options
+
+  const tempFile = `${output}.tmp`
+
+  return new Promise((resolve, reject) => {
+    const readStream = fs.createReadStream(output, { encoding: 'utf-8' })
+    const writeStream = fs.createWriteStream(tempFile, { encoding: 'utf-8' })
```

Comment:
I was wondering if we should maybe use [`os.tmpdir`](https://nodejs.org/api/os.html#os_os_tmpdir) to generate the temporary file. And there is also [`fs.mkdtemp`](https://nodejs.org/api/fs.html#fs_fs_mkdtemp_prefix_options_callback)

Reply:
@bpetetot can we look at this in an other PR?

I just tried to do it quickly and here what I got:
```
✖  error     Error: EXDEV: cross-device link not permitted, rename '/var/folders
/sn/k62l9zk12rscm8c44zdmfm_00000gn/T/gitmoji-cli8mJAZ9/CHANGELOG.md.tmp' -> './C
HANGELOG.md'
```

I think it opens new kind of problems.

And I was wondering if this is not better to do this in memory (without streams) since CHANGELOG files are not > 10M I guess...

Reply:
No problem to do that in another PR.